trunk/SUAVE/Components/Energy/Networks/Internal_Combustion_Propeller.py

```diff
@@ -0,0 +1,153 @@
+            None
+    
+            Source:
+            N/A
+    
+            Inputs:
+            None
+    
+            Outputs:
+            None
+    
+            Properties Used:
+            N/A
+        """        
+        self.engine            = None
+        self.propeller         = None
+        self.engine_length     = None
+        self.number_of_engines = None
+        self.thrust_angle      = 0.0
+        self.rated_speed       = 0.0
+    
+    # manage process with a driver function
+    def evaluate_thrust(self,state):
+        """ Calculate thrust given the current state of the vehicle
+    
+            Assumptions:
+    
+            Source:
+            N/A
+    
+            Inputs:
+            state [state()]
+    
+            Outputs:
+            results.thrust_force_vector [newtons]
+            results.vehicle_mass_rate   [kg/s]
+            conditions.propulsion:
+                rpm                  [radians/sec]
+                propeller_torque     [N-M]
+                power                [W]
+    
+            Properties Used:
+            Defaulted values
+        """           
+        # unpack
+        conditions  = state.conditions
+        numerics    = state.numerics
+        engine      = self.engine
+        propeller   = self.propeller
+        rated_speed = self.rated_speed
+        
+        # Throttle the engine
+        eta = conditions.propulsion.throttle[:,0,None]
+        engine.speed = rated_speed * eta
+        conditions.propulsion.combustion_engine_throttle = eta # keep this 'throttle' on
+        
+        # Run the engine
+        engine.power(conditions)
+        power_output = engine.outputs.power
+        sfc          = engine.outputs.power_specific_fuel_consumption
+        mdot         = engine.outputs.fuel_flow_rate
+        torque       = engine.outputs.torque     
+        
+        # link
+        propeller.inputs.omega = engine.speed
+        propeller.thrust_angle = self.thrust_angle
+        
+        # step 4
+        F, Q, P, Cp ,  outputs  , etap  = propeller.spin(conditions)
+        
+        # Check to see if magic thrust is needed, the ESC caps throttle at 1.1 already
+        P[eta>1.0] = P[eta>1.0]*eta[eta>1.0]
+        F[eta>1.0] = F[eta>1.0]*eta[eta>1.0]   
+        
+        # link
+        propeller.outputs = outputs
+    
+        # Pack the conditions for outputs
+        rpm        = engine.speed / Units.rpm
+          
+        conditions.propulsion.rpm              = rpm
+        conditions.propulsion.propeller_torque = Q
+        conditions.propulsion.power            = P
+        
+        # Create the outputs
+        F    = self.number_of_engines * F * [np.cos(self.thrust_angle),0,-np.sin(self.thrust_angle)]      
+        
+        results = Data()
+        results.thrust_force_vector = F
+        results.vehicle_mass_rate   = mdot
+        
+        return results
+    
+    
+    def unpack_unknowns(self,segment,state):
+        """"""        
```

Comment:
Unfinished docstrings here and in the next function.

Reply:
Docstrings added